replica/table.cc

```diff
@@ -2035,6 +2070,29 @@ future<> table::generate_and_propagate_view_updates(shared_ptr<db::view::view_up
         }
         tracing::trace(tr_state, "Generated {} view update mutations", updates->size());
         auto units = seastar::consume_units(*_config.view_update_concurrency_semaphore, memory_usage_of(*updates));
+
```

Comment:
Instead of this consume_units, why not also wait for units to become available?

btw, this should be separate for system/user (and service level) to prevent priority inversion.

Reply:
There's the idea that view updates shouldn't block normal queries, to maintain high availability, so we shouldn't wait on any semaphores. When there's no resources in the semaphore we should just shed the load. Throttling should make sure that this never happens (in a workload with limited concurrency). I'm not sure how valid this idea is, I tried to change it in another PR, but there was some backpressure against this idea.

/cc @denesb

Reply:
Once the server is overloaded, we have two options: throttle normal queries, or drop view updates. Of the two, throttling normal queries is much better.

Just sleeping without throttling the real client will mean we'll have more and more sleepers, until this node's memory is consumed by sleeping view updates.

Reply:
The real client is throttled as well because the view backlog increases as the view updates are generated.
With large backlog, all normal client writes will be throttled as well.

The idea of sleep-based throttling is that it will eventually sleep enough to ensure that the amount of view updates doesn't increase anymore. The concurrency is bounded, but in an indirect way.

Reply:
Can we prove that this indeed happens?